src/parser/nscontext.js

```diff
@@ -206,7 +206,7 @@ class NamespaceContext {
         return mapping;
       }
     }
-    if (this.getNamespaceURI(prefix, true)) {
+    if (this.getNamespaceURI(prefix)) {
```

Comment:
I wonder why you need to check the prefix beyond the local scope. We only have to generate a unique prefix if there is a conflict at current level. It's fine to redeclare a prefix mapping to a different uri.